### ISSUE TITLE
FNX-4541 ⁃ [Bug]Native code crash on Beta 5.1.0-beta.1 after accepting the cookie policy on Theverge.com

### ISSUE THREAD

**Issue by AndiAJ** [mozilla-mobile/fenix#10770]
## Steps to reproduce
1. Navigate to theverge.com
1. Accept the cookie policy

### Expected behavior
Page is properly displayed after reload

### Actual behavior
Native code crash

### Device information
* Android device: OnePlus A3 (Android 6.0.1)
* Fenix version: enix version: Beta 5.1.0-beta.1 from 5/19

### Notes
► Crash report link https://crash-stats.mozilla.org/report/index/1aa99422-2ff5-47b2-9f3e-2daa80200519

❗ Didn't manage to reproduce on the following devices:
• Google Pixel 3a (Android 10)
• Huawei Mate 20 Lite (Android 9)

❗ Didn't manage to reproduce on the latest **`Nightly Build 200519 from 5/19`**

►Video
![20200519-161947](https://user-images.githubusercontent.com/51314259/82332794-83292780-99ee-11ea-98f0-ca80718a0aeb.gif)

► Log
```kotlin
05-19 16:20:06.908 1302-2216/? I/ActivityManager: Process org.mozilla.fenix.beta:tab0 (pid 11108) has died
05-19 16:20:06.909 1302-2216/? W/ActivityManager: Scheduling restart of crashed service org.mozilla.fenix.beta/org.mozilla.gecko.process.GeckoChildProcessServices$tab0 in 1000ms
05-19 16:20:06.918 11060-11084/? W/GeckoEventDispatcher: No listener for GeckoView:WebExtension:Disconnect
05-19 16:20:06.921 11060-11060/? D/GeckoNetworkManager: Incoming event disableNotifications for state OnWithListeners -> OnNoListeners
05-19 16:20:06.925 720-720/? I/Zygote: Process 11108 exited due to signal (11)
05-19 16:20:06.925 1302-1342/? V/ActivityManager: killProcessGroupAsync took 16 ms for PID 11108 on thread 14
05-19 16:20:06.938 11060-11095/? I/Gecko: [Parent 11060, Unnamed thread 7f70744240] WARNING: waitpid failed pid:11108 errno:10: file /builds/worker/checkouts/gecko/ipc/chromium/src/base/process_util_posix.cc, line 237
05-19 16:20:06.947 1302-2209/? I/ActivityManager: Start proc 11399:org.mozilla.fenix.beta:mozilla.components.lib.crash.CrashHandler/u0a286 for service org.mozilla.fenix.beta/mozilla.components.lib.crash.handler.CrashHandlerService
05-19 16:20:07.171 11060-11060/? W/GeckoSession: Attempted to close a GeckoSession that was already closed.
05-19 16:20:07.174 11060-11060/? I/art: Note: end time exceeds epoch: 
05-19 16:20:07.181 11060-11198/? I/Gecko: [GFX1-]: Failed to create EGLContext!: 0x300c
05-19 16:20:07.182 11060-11060/? W/GeckoEventDispatcher: No listener for GeckoView:AccessibilityEnabled
05-19 16:20:07.209 11399-11413/? E/mozac/CrashReporter: CrashHandlerService received native code crash
05-19 16:20:07.212 11399-11413/? I/mozac/CrashReporter: Received crash: NativeCodeCrash(minidumpPath=/data/user/0/org.mozilla.fenix.beta/files/mozilla/Crash Reports/pending/5ab2f27f-a13b-ce9c-6e1f-4f08c8b89445.dmp, minidumpSuccess=false, extrasPath=/data/user/0/org.mozilla.fenix.beta/files/mozilla/Crash Reports/pending/5ab2f27f-a13b-ce9c-6e1f-4f08c8b89445.extra, isFatal=false, breadcrumbs=[], uuid=8c92fbb0-04b0-415a-9a7c-373153a67388)
05-19 16:20:07.261 1302-1810/? I/ActivityManager: Start proc 11414:org.mozilla.fenix.beta:tab0/u0a286 for service org.mozilla.fenix.beta/org.mozilla.gecko.process.GeckoChildProcessServices$tab0
05-19 16:20:07.263 11060-11102/? D/ServiceAllocator: org.mozilla.gecko.process.GeckoChildProcessServices$tab0 updateBindings: BACKGROUND priority, 0 importance, 2 successful binds, 0 failed binds, 0 successful unbinds, 0 failed unbinds
05-19 16:20:07.268 11399-11413/? I/mozac/CrashReporter: Invoking non-fatal PendingIntent
05-19 16:20:07.269 1302-2248/? I/ActivityManager: START u0 {flg=0x14000000 cmp=org.mozilla.fenix.beta/org.mozilla.fenix.HomeActivity (has extras)} from uid 10286 pid -1 on display 0
05-19 16:20:07.272 11060-11130/? I/mozac/CrashReporter: Crash report submitted to 1 telemetry services
05-19 16:20:07.279 11399-11413/? I/art: System.exit called, status: 0
05-19 16:20:07.279 11399-11413/? I/AndroidRuntime: VM exiting with result code 0, cleanup skipped.
05-19 16:20:07.316 1302-2064/? I/ActivityManager: Process org.mozilla.fenix.beta:mozilla.components.lib.crash.CrashHandler (pid 11399) has died
05-19 16:20:07.317 1302-2064/? W/ActivityManager: Scheduling restart of crashed service org.mozilla.fenix.beta/mozilla.components.lib.crash.handler.CrashHandlerService in 1000ms
05-19 16:20:07.320 1302-1729/? D/ConnectivityService: releasing NetworkRequest NetworkRequest [ id=367, legacyType=-1, [ Transports: WIFI Capabilities: NOT_RESTRICTED&TRUSTED&NOT_VPN] ]
05-19 16:20:07.327 1302-1342/? V/ActivityManager: killProcessGroupAsync took 0 ms for PID 11399 on thread 14
05-19 16:20:07.332 11060-11126/? W/FirefoxAccountStateMach: Got invalid event Event$Init for state NotAuthenticated.
05-19 16:20:07.335 11060-11060/? D/PhoneWindow: state.preset = true, color: 0, translucent = 0
05-19 16:20:07.480 11414-11414/? D/GeckoThread: State changed to LAUNCHED
05-19 16:20:07.482 11414-11429/? I/GeckoThread: preparing to run Gecko```

**Comment by fluffyemily** [comment]
GV issue duped to fixed bug https://bugzilla.mozilla.org/show_bug.cgi?id=1638016. We may have to uplift to get this bug in beta.

**Comment by liuche** [comment]
Clicked through the bugzilla links, and the GV issue above is duped to an open bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1635720

**Comment by liuche** [comment]
I'm going through [our crash-stats graphs](https://crash-stats.mozilla.org/signature/?product=Fenix&signature=mozilla%3A%3ABufferList%3CT%3E%3A%3AIterImpl%3A%3AData&date=%3E%3D2020-02-19T20%3A16%3A00.000Z&date=%3C2020-05-19T20%3A16%3A00.000Z#graphs) for this specific signature, to see if I can see how many/how often this is crashing.

Looks like it's not all devices, started in 4.3 and continuing in 5.0.

Thank you so much for linking to this crash report!

**Comment by dblohm7** [comment]
Note for the STR: It looks like The Verge only shows the cookie policy when connecting from a jurisdiction where this is required. When I load it from my location, I don't see it.

I'll try going through my VPN to see if that helps...

**Comment by dblohm7** [comment]
I was able to receive the cookie policy notice, however I did not see any crashes after accepting. :-(

I should note that after accepting cookies, ads start showing. I wonder if it's specific ad content that could be triggering this.

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on **`Beta 5.1.0-beta.2 from 5/27`** using a OnePlus A3 (Android 6.0.1)

► Video
![20200527-144300](https://user-images.githubusercontent.com/51314259/83016105-1c2cf380-a02a-11ea-9641-453c9589497f.gif)
